src/etc/inc/plugins.inc.d/webgui.inc

```diff
@@ -66,13 +66,20 @@ function webgui_configure_do($verbose = false, $interface = '')
 
     $listeners = count($interfaces) ? array() : array('0.0.0.0', '::');
 
-    foreach (interfaces_addresses($interfaces) as $tmpaddr => $info) {
-        if ($info['scope']) {
-            /* link-local does not seem to be supported */
-            continue;
-        }
+    /* bind on (all/selected) interfaces or use specified IP */
+    if (isset($config['system']['webgui']['managementaccess']) &&
+        $config['system']['webgui']['managementaccess'] === 'ip' &&
+        isset($config['system']['webgui']['ip'])) {
+        $listeners[] = $config['system']['webgui']['ip'];
```

Comment:
```suggestion
        $listeners = array($config['system']['webgui']['ip']);
```

Otherwise the management IP is actually **added** to the array, which already contains `0.0.0.0` and `::` in case the setting `webguiinterfaces` (which is hidden in the UI when the management IP is configured) is set to "all". This would again result in listening to all IPs of the system.

Reply:
It works as intended. It _adds_ the IP to the array, but there's nothing else in the array.

Reply:
@fraenki The setting `webguiinterfaces` (which can have one interface or none (none means "all" in the GUI) is still relevant, even when your new setting `managementaccess` is set. That means if you configured "none" on the interfaces, then enable your new setting and configure a management IP, the array contains `['0.0.0.0', '::']` due to line 67.

### Detailed steps to reproduce:
1. Configure _Management Access: Interface_ and _Listen Interfaces: All_
2. Save
3. Change _Management Access_ to _IP_ and configure a _Management IP_
4. Save
5. Result: 
```
$SERVER["socket"] == "0.0.0.0:443" {
    ssl.engine = "enable"
}
$SERVER["socket"] == "[::]:443" {
    ssl.engine = "enable"
}
$SERVER["socket"] == "10.8.16.4:443" {
    ssl.engine = "enable"
}
```
```
root@opnsense-dev ~/core$ sockstat -l -p 443 -P tcp                                                                                         ✹ ✭fraenki-mgmt_ip
USER     COMMAND    PID   FD PROTO  LOCAL ADDRESS         FOREIGN ADDRESS
root     lighttpd   28935 5  tcp4   *:443                 *:*
root     lighttpd   28935 6  tcp6   *:443                 *:*
root     lighttpd   28935 7  tcp4   10.8.16.4:443         *:*
```

Reply:
@8191 Thanks for the detailed explanation.